toolchain/sem_ir/entity_with_params_base.h

```diff
@@ -95,21 +94,23 @@ struct EntityWithParamsBase {
   // A block containing the pattern insts for the parameter lists.
   InstBlockId pattern_block_id;
   // A block containing, for each implicit parameter, a reference to the
-  // instruction in the entity's declaration block that depends on all other
-  // pattern-match insts pertaining to that parameter.
-  InstBlockId implicit_param_refs_id;
-  // A block containing, for each implicit parameter, a reference to the
   // instruction in the entity's pattern block that depends on all other
   // pattern insts pertaining to that parameter.
   InstBlockId implicit_param_patterns_id;
   // A block containing, for each explicit parameter, a reference to the
-  // instruction in the entity's declaration block that depends on all other
-  // pattern-match insts pertaining to that parameter.
-  InstBlockId param_refs_id;
-  // A block containing, for each explicit parameter, a reference to the
   // instruction in the entity's pattern block that depends on all other
   // pattern insts pertaining to that parameter.
   InstBlockId param_patterns_id;
+  // A block containing, for each calling convention parameter, a reference to
+  // the instruction in the function's declaration block that represents that
+  // parameter. These parameters appear in declaration order: `self` (if
+  // present), then the explicit runtime parameters, then the return slot (which
```

Comment:
Right now, the only implicit parameter the toolchain supports is `self`. However, once more are supported, won't those become part of the calling convention? Also, are you intending to exclude non-runtime explicit parameters (do we have those?)?

You might consider something like:

```suggestion
  // parameter. Parameters are in declaration order: implicit parameters, explicit parameters, then the return slot (which
```

Reply:
I'm intending to exclude all non-runtime parameters, because they aren't part of the calling convention (i.e. they aren't passed as part of the function call). Unless I'm missing something, `self` is the only non-compile-time parameter that can appear in the implicit parameter list.